projects/core-infra/src/components/identity-pool-github.ts

```diff
@@ -26,7 +27,7 @@ export class IdentityPoolGithubSetup extends pulumi.ComponentResource {
     new GithubIdentityPoolIamMember(
       name,
       { repo, owner, serviceAccountId },
-      { parent: this },
+      { parent: this, provider: githubProvider },
```

Comment:
```suggestion
      { parent: this },
```

Providers for Pulumi components should be defined when they are constructed.